File: app/src/main/java/com/divercity/app/features/home/jobs/JobsFragment.java
package com.divercity.app.features.home.jobs;


import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProviders;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.View;

import com.divercity.app.BR;
import com.divercity.app.R;
import com.divercity.app.core.base.BaseFragment;
import com.divercity.app.databinding.FragmentJobsBinding;
import com.divercity.app.features.home.HomeActivity;

import javax.inject.Inject;

/**
 * A simple {@link Fragment} subclass.
 */
public class JobsFragment extends BaseFragment {

    private JobsViewModel viewModel;
    private FragmentJobsBinding binding;

    @Inject
    JobsViewPagerAdapter adapter;

    public JobsFragment() {
        // Required empty public constructor
    }

    @Override
    public int getLayoutId() {
        return R.layout.fragment_jobs;
    }

    @Override
    public ViewModel getViewModel() {
        viewModel = ViewModelProviders.of(getActivity(), viewModelFactory).get(JobsViewModel.class);
        return viewModel;
    }

    @Override
    public int getBindingVariable() {
        return BR.viewModel;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        binding = (FragmentJobsBinding) getViewDataBinding();
        setupToolbar();
        binding.viewpager.setAdapter(adapter);
        binding.tabLayout.setupWithViewPager(binding.viewpager);
    }

    private void setupToolbar() {
        binding.includeToolbar.title.setText(getResources().getString(R.string.divercity));
        ((HomeActivity) getActivity()).setSupportActionBar(binding.includeToolbar.toolbar);
        ((HomeActivity) getActivity()).getSupportActionBar().setDisplayShowTitleEnabled(false);
        setHasOptionsMenu(true);
    }

    @Override
    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        inflater.inflate(R.menu.menu_jobs, menu);
        super.onCreateOptionsMenu(menu, inflater);
    }
}

File: app/src/main/java/com/divercity/app/features/home/jobs/JobsFragment.kt
package com.divercity.app.features.home.jobs

import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.view.View
import com.divercity.app.R
import com.divercity.app.core.base.BaseFragment
import com.divercity.app.features.home.HomeActivity
import kotlinx.android.synthetic.main.fragment_jobs.*
import javax.inject.Inject

/**
 * Created by lucas on 25/10/2018.
 */


class JobsFragment : BaseFragment() {

    lateinit var viewModel: JobsViewModel

    @Inject
    lateinit var adapter: JobsViewPagerAdapter

    companion object {

        fun newInstance(): JobsFragment {
            return JobsFragment()
        }
    }

    override fun layoutId(): Int = R.layout.fragment_jobs

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        viewModel = activity?.run {
            ViewModelProviders.of(this, viewModelFactory).get(JobsViewModel::class.java)
        } ?: throw Exception("Invalid Fragment")

        setupToolbar()
        setupAdapterViewPager()
        subscribeToLiveData()
    }

    private fun subscribeToLiveData() {
        viewModel.showBtnAddJob.observe(this, Observer { data ->
            data?.let {
                (btn_add as View).visibility = data
                btn_add.setOnClickListener {
                    navigator.navigateToJobPostingActivity(activity!!)
                }
            }
        })
    }

    private fun setupToolbar() {
        (activity as HomeActivity).apply {
            supportActionBar?.let {
                it.setTitle(R.string.job_board)
                it.setDisplayHomeAsUpEnabled(false)
            }
        }
    }

    private fun setupAdapterViewPager() {
        viewpager.adapter = adapter
        tab_layout.setupWithViewPager(viewpager)
    }
}
